### PR TITLE
Expand information about testing in `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by delucis**
This PR adds some of what I learned from @hippotastic in #38 to the testing section of the contributing guide.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -52,6 +52,36 @@ The root package includes a helpful shortcut that you can use to build all packa
...
+
+You can use the `test-watch` command to start a live reloading dev process that re-runs tests each time code changes. This also works with the `--filter` flag to only run tests for a specific package:
+
+```sh
+# Run tests for all packages and watch for changes:
+pnpm -r test-watch
+
+# Run tests for frames plugin and watch for changes:
+pnpm --filter plugin-frames test-watch
+```
```
Comment (hippotastic) [inline CONTRIBUTING.md:79]:
````suggestion
### Live reloading tests

You can use the `test-watch` command provided by the individual packages to start a live reloading dev process that re-runs tests each time code changes.

To select the package you want to run tests for, you have two options:

- Change your working directory to one of the packages and run the command directly:

  ```sh
  # Change to the frames plugin directory:
  cd packages/@expressive-code/plugin-frames

  # Run tests for the current package and watch for changes:
  pnpm test-watch
  ```

- Alternatively, stay in the repository root and use the `--filter` flag to run tests for a specific package:

  ```sh
  # Run tests for the frames plugin and watch for changes:
  pnpm --filter plugin-frames test-watch
  ```
````

Unfortunately, `pnpm -r test-watch` doesn't work for me. It only runs the `test-watch` command of the first package (usually `core`) and waits for changes. I've therefore changed the text to reflect that `test-watch` is only available on individual packages, and to show both ways of running the command.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -52,6 +52,36 @@ The root package includes a helpful shortcut that you can use to build all packa
...
+
+### Visualizing changes
+
+Tests output HTML files to `test/__html_snapshots__/` directories. You can open these files in a browser after running tests to see the impact of your changes.
+
```
Comment (hippotastic) [inline CONTRIBUTING.md:84]:
```suggestion

You can even combine live reloading tests (see above) with VS Code's [Live Preview](https://marketplace.visualstudio.com/items?itemName=ms-vscode.live-server) extension, which adds a `Show Preview` context menu item to all HTML files. Run the `test-watch` command, open a preview of any of the generated HTML snapshots, and the effects of your code changes will be visible in a live-updating preview right in your editor.

```

I'd suggest adding this powerful combination as well. I'm using this all the time while I'm making changes.

**Review by hippotastic — COMMENTED** [review]
This is great, thank you very much!

I made two suggestions, please have a look. 😊

File: CONTRIBUTING.md (hunk 1 above)
Comment (delucis) [reply CONTRIBUTING.md:79]: Ah yes, you’re right. I think I tested `pnpm -r test` and didn’t pay as close attention to the output when doing the same for the watch version.

File: CONTRIBUTING.md (hunk 2 above)
Comment (delucis) [reply CONTRIBUTING.md:95]: Agreed — this is exactly what I ended up doing too.

**Review by hippotastic — APPROVED** [review]
Great work, thank you very much!

**Pull request merged**